capsules/extra/src/analog_sensor.rs

```diff
@@ -82,9 +82,9 @@ impl<'a, A: hil::adc::Adc<'a>> AnalogTemperatureSensor<'a, A> {
     pub fn new(
         adc: &'a A,
         channel: &'a <A as kernel::hil::adc::Adc<'a>>::Channel,
-        sensor_type: AnalogLightSensorType,
-    ) -> AnalogLightSensor<'a, A> {
-        AnalogLightSensor {
+        sensor_type: AnalogTemperatureSensorType,
+    ) -> Self {
+        Self {
```

Comment:
This seems like it fixes a bug(?) where `sensor_type` had an incorrect type? Should probably be its own PR.